mica/starcheck/starcheck.py

```diff
@@ -8,90 +8,234 @@
+                      timelines_db=dict(dbi='sqlite',
+                                        server=os.path.join(os.environ['SKA'], 'data', 'cmd_states', 'cmd_states.db3')))
 FILES = dict(data_root=os.path.join(MICA_ARCHIVE, 'starcheck'),
              touch_file=os.path.join(MICA_ARCHIVE, 'starcheck', "starcheck_parser.touch"),
              sql_def='starcheck.sql')
 
-# this doesn't belong in starcheck
-def get_timeline_at_date(date):
-    aca_db = Ska.DBI.DBI(dbi='sybase', server='sybase', user='aca_read')
-    return aca_db.fetchone(
-        "select * from timeline_loads where datestop >= '%s' "
-        " and datestart <= '%s' and scs <= 130 order by datestart desc"
-        % (date, date))
 
-def get_mp_dir(obsid, config=None):
+def get_monitor_windows(start=None, stop=None, min_obsid=40000, config=None):
+    """
+    Use the database of starcheck products to get a list of monitor windows
+    This list is filtered by timelines content to only include catalogs that should have or will run.
+
+    :param start: Optional start time for filtering windows as fetched from the database
+    :param stop: Optional stop time for filtering windows as fetched from the database
+    :param min_obsid: Minimum obsid value for filtering.  Default of 40000 is intended to fetch only ERs
+    :param config: config dictionary. If supplied must include 'starcheck_db' and 'timelines_db' keys
+                   with dictionaries of the required arguments to Ska.DBI to connect to those databases.
+    :returns: astropy Table of monitor windows.  See get_starcheck_catalog_at_date for description of the values
+              of the 'status' column.  The 'catalog' column contains the get_starcheck_catalog_at_date
+              returned dictionary.
+    """
     if config is None:
         config = DEFAULT_CONFIG
-    aca_db = Ska.DBI.DBI(dbi='sybase', server='sybase', user='aca_read')
-    possible_runs = aca_db.fetchall(
-        "select * from tl_obsids where obsid = %d" % obsid)
-    if not len(possible_runs):
-        db = Ska.DBI.DBI(dbi='sqlite', server=config['server'])
-        sc = db.fetchall(
+    start_date = DateTime(start or '1999:001').date
+    stop_date = DateTime(stop).date
+    with Ska.DBI.DBI(**config['starcheck_db']) as db:
+        mons = db.fetchall("""select obsid, mp_starcat_time as mp_starcat_date, type, sz, yang, zang, dir
+                          from starcheck_catalog, starcheck_id
+                          where type = 'MON' and obsid > {}
+                          and mp_starcat_date >= '{}'
+                          and mp_starcat_date <= '{}'
+                          and starcheck_id.id = starcheck_catalog.sc_id""".format(
+                min_obsid, start_date, stop_date))
+    mons = Table(mons)
+    mons['catalog'] = None
+    with Ska.DBI.DBI(**config['timelines_db']) as timelines_db:
+        statuses = []
+        # Check which ones actually ran or are likely to run
+        for i_mon, mon in enumerate(mons):
```

Comment:
Oops, looks like enumerated but didn't wait to commit this before finding out from @taldcroft why that was suggested as part of the strategy to add the catalogs to the data structure.